tasks/main.yml

```diff
@@ -1,90 +1,101 @@
 ---
-# datasource.0 = grafana_datasource attribute
-- name: Manage datasource
-  community.grafana.grafana_datasource:
+- name: Grafana connection anchor
+  ansible.builtin.set_fact: &grafana_connection
```

Comment:
I don't really see the benefit in using a yaml-anchor here. Yes, we save some lines, but then again we have a new task (that would need a `always`-tag) and a yaml-functionality that is not widely used (IMO). So I prefer using the variant without the anchors.

If you really want to save some lines, I'd prefer using this solution: https://stackoverflow.com/a/54466785

Reply:
Unfortunately, `module_defaults` doesn’t make much sense, because each task uses its own module.
Defining `action_groups` in `meta/runtime.yml` is also not an option as this is only available for collections.
Either we leave the anchor, or the url-parameters are defined on every task. For now I would rollback to definition per task.